newrelic/hooks/mlmodel_openai.py

```diff
@@ -406,66 +205,55 @@ async def wrap_embedding_async(wrapped, instance, args, kwargs):
     embedding_id = str(uuid.uuid4())
 
-    span_id = None
-    trace_id = None
-
-    function_name = wrapped.__name__
-
-    with FunctionTrace(name=function_name, group="Llm/embedding/OpenAI") as ft:
-        # Get trace information
-        available_metadata = get_trace_linking_metadata()
-        span_id = available_metadata.get("span.id", "")
-        trace_id = available_metadata.get("trace.id", "")
-
+    with FunctionTrace(name=wrapped.__name__, group="Llm/embedding/OpenAI") as ft:
+        linking_metadata = get_trace_linking_metadata()
         try:
             response = await wrapped(*args, **kwargs)
         except Exception as exc:
-            _record_embedding_error(transaction, embedding_id, span_id, trace_id, kwargs, ft, exc)
+            _record_embedding_error(transaction, embedding_id, linking_metadata, kwargs, ft, exc)
             raise
 
     if not response:
         return response
 
-    _record_embedding_success(transaction, embedding_id, span_id, trace_id, kwargs, ft, response)
+    _record_embedding_success(transaction, embedding_id, linking_metadata, kwargs, ft, response)
     return response
 
 
-def _record_embedding_success(transaction, embedding_id, span_id, trace_id, kwargs, ft, response):
+def _record_embedding_success(transaction, embedding_id, linking_metadata, kwargs, ft, response):
     settings = transaction.settings if transaction.settings is not None else global_settings()
-    # Grab LLM-related custom attributes off of the transaction to store as metadata on LLM events
-    custom_attrs_dict = transaction._custom_params
-    llm_metadata_dict = {key: value for key, value in custom_attrs_dict.items() if key.startswith("llm.")}
-
+    span_id = linking_metadata.get("span.id")
+    trace_id = linking_metadata.get("trace.id")
     response_headers = getattr(response, "_nr_response_headers", {})
 
-    # In v1, response objects are pydantic models so this function call converts the object back to a dictionary for backwards compatibility
-    # Use standard response object returned from create call for v0
+    # In v1, response objects are pydantic models so this function call converts the
+    # object back to a dictionary for backwards compatibility.
+    attribute_response = response
     if OPENAI_V1:
         attribute_response = response.model_dump()
-    else:
-        attribute_response = response
 
-    request_id = response_headers.get("x-request-id", "") if response_headers else ""
-
-    response_model = attribute_response.get("model", "") or ""
+    request_id = response_headers.get("x-request-id")
+    response_model = attribute_response.get("model")
     response_usage = attribute_response.get("usage", {}) or {}
-    api_type = getattr(attribute_response, "api_type", "")
-    organization = response_headers.get("openai-organization", "") if OPENAI_V1 else attribute_response.organization
+    api_type = getattr(attribute_response, "api_type", None)
```

Comment:
Looking at this again, we may honestly be able to remove `api_type`. It's not something I listed in the spec for other agents and it's a legacy attribute at this point anyway since v1 doesn't report it.